pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -121,19 +134,12 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 	if err != nil {
 		return err
 	}
-	if platform == PlatformAWS {
-		r.reqLogger.Info("not creating a CephObjectStore because the platform is AWS")
-		return nil
-	}
-	if platform == PlatformGCP {
-		r.reqLogger.Info("not creating a CephObjectStore because the platform is GCP")
+	if isCloudPlatform(platform) {
+		reqLogger.Info(fmt.Sprintf("not creating a CephObjectStore because the platform is '%s'", platform))
+		// as the platform is not supported, we should not re-enter the function
+		instance.Status.CephObjectStoresCreated = true
```

Comment:
Logger says we are not creating CephObjectStore. Why does the status say `True` (i.e Created)?

Reply:
If you don't put the status true, the code will re-enter the function (till this point) every time operator reconciles and we don't want to do that...

Reply:
So we need to change when/how reconcile is triggered. We should not be reporting false status. This is something the user can see.

Reply:
Ok... I have removed that line, but again just want to bring the attention back to the multiple call of `r.platform.GetPlatform()` function in the module. The function will be called for every reconcile, even when we realized (in the previous attempt itself) that platform is non-usable.

Addressing that issue can be go in as a performance bug fix, and should not block this PR.

Reply:
Added back the original `rgw` placement with changes required for the anti-affinity (into the `Changes for 'GatewaySpec', required for a multi-instances` commit)...
@umangachapagain , please take a look...